src/components/reusable/progressBar/progressBar.ts

```diff
@@ -0,0 +1,273 @@
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+import '../loaders';
+import '../tooltip';
+import checkmarkIcon from '@carbon/icons/es/checkmark--filled/16';
+import errorIcon from '@carbon/icons/es/error--filled/16';
+import informationIcon from '@carbon/icons/es/information/16';
+
+import ProgressBarStyles from './progressBar.scss';
+
+enum ProgressStatus {
+  INDETERMINATE = 'indeterminate',
+  ACTIVE = 'active',
+  SUCCESS = 'success',
+  ERROR = 'error',
+}
+
+/**
+ * `<kyn-progress-bar>` -- progress bar status indicator component.
+ */
+@customElement('kyn-progress-bar')
+export class ProgressBar extends LitElement {
+  static override styles = ProgressBarStyles;
+
+  /** Sets visibility of optional inline load status spinner. */
+  @property({ type: Boolean })
+  showInlineLoadStatus = true;
+
+  /** Sets progress bar status mode. */
+  @property({ type: String })
+  status: 'indeterminate' | 'active' | 'success' | 'error' = 'indeterminate';
+
+  /** Sets initial progress bar value (optionally hard-coded). */
+  @property({ type: Number })
+  value: number | null = null;
+
+  /** Sets manual max value (default = 100). */
+  @property({ type: Number })
+  max = 100;
+
+  /** Sets optional progress bar label. */
+  @property({ type: String })
+  label = '';
+
+  /** Sets optionally displayed information icon and tooltip content. */
+  @property({ type: String })
+  informationalTooltipText = '';
+
+  /** Sets optional helper text that appears underneath `<progress>` element. */
+  @property({ type: String })
+  helperText = '';
+
+  /** Sets the unit for progress measurement (ex: 'MB', 'GB', '%') */
+  @property({ type: String })
+  unit = '';
+
+  /** Incrementing percentage count value.
+   * @internal
+   */
+  @state()
+  private _percentage = 0;
+
+  /** Increments animated movement in progress bar.
+   * @internal
+   */
+  @state()
+  private _progress = 0;
+
+  /** Value indicates whether or not the bar is in animated motion, dynamic helper text values are proportionally incrementing.
+   * @internal
+   */
+  @state()
+  private _running = false;
+
+  /** Controls timeout interval for incremented bar animation.
+   * @internal
+   */
+  private _animationFrameId: number | null = null;
+
+  override render() {
+    const currentValue = this.status === 'active' ? this._progress : this.value;
+    const currentStatus = this.getCurrentStatus(currentValue);
+    const helperText = this.getHelperText();
+
+    this._percentage = Math.round((this._progress / this.max) * 100);
+
+    return html`
+      ${this.renderProgressBarLabel(currentStatus)}
+      ${this.renderProgressBar(currentStatus, currentValue)}
+      ${helperText
+        ? html`<div class=${`progress-bar__helper-text ${currentStatus}`}>
+            <h2>${helperText}</h2>
```

Comment:
Don't bake-in any heading tags, since the component won't be aware of the overall page hierarchy.